### PR TITLE
Add "Brown + Pink waves" sound

### LINKED ISSUE #842: Add wavy brown+pink noise
Personally, I like the monotonous brown noise, but for sleeping it's better to listen to its [wavy version](https://github.com/ashutoshgngwr/noice/files/7447955/sox-waves-edit2.zip). Regular crashing waves are too crispy.

If you'd like to generate it (as the other raw noises) instead of re-using the sample, you need to use [SoX](http://sox.sourceforge.net/).
After installed, it's played with this command:
`play -n synth brownnoise synth pinknoise mix synth 0 0 0 10 10 40 trapezium amod 0.1 30`
Or recorded with this:
`sox -n -r 44100 output.wav synth 60 brownnoise synth 60 pinknoise mix synth 0 0 0 10 10 40 trapezium amod 0.1 30`
(where `60` is length in seconds, it's better to generate initially in wav, then to mp3 via ffmpeg)

Originally the command was written on [AskUbuntu](https://askubuntu.com/questions/789465/generate-white-noise-to-calm-a-baby/789472#comment1296847_789472).
```
The longer period [10 seconds] makes it harder to notice the regularity. 
The numbers mean: 
0.1 = wave crash frequency (0.1 Hz = one wave every 10 seconds); 
30% = noise level in between waves; 
1st 10% = duration (over the 10 seconds) of the rising part of the wave; 
2nd 10% = when it starts falling 
(they are both 10% so that the wave crashes as soon as it's done rising in volume); 
40% duration of the entire wave: 
this means 100% - 40% = 60% of the time is spent waiting between waves.
– Tobia
```

The reason I'd like to see this in Noice is because the app provides smooth turn-off when the timer goes off, which can't be achieved as easily with just SoX.

### PR DESCRIPTION
**Pull request by Rult**
Closes #842.

### CODE REVIEW

File: app/src/main/java/com/github/ashutoshgngwr/noice/model/Sound.kt [hunk 1]
```diff
@@ -64,6 +64,12 @@ class Sound private constructor(
         displayGroupResID = R.string.sound_group__raw_noise,
         iconID = R.drawable.ic_sound_brownian_noise,
       ),
+      "brown_plus_pink_waves" to Sound(
+        src = arrayOf("brown_plus_pink_waves.mp3"),
+        titleResID = R.string.brown_plus_pink_waves,
+        displayGroupResID = R.string.sound_group__raw_noise,
+        iconID = R.drawable.ic_sound_seaside,
```
Comment (ashutoshgngwr) [inline app/src/main/java/com/github/ashutoshgngwr/noice/model/Sound.kt:71]: TODO: update the icon

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -78,6 +78,7 @@
     <string name="white_noise">White noise</string>
     <string name="wind_in_chimes_of_shells">Wind in chimes of shells</string>
     <string name="wind_in_palm_trees">Wind in palm trees</string>
+    <string name="brown_plus_pink_waves">Brown+Pink waves</string>
```
Comment (ashutoshgngwr) [inline app/src/main/res/values/strings.xml:81]:
```suggestion
    <string name="brown_plus_pink_waves">Brown + Pink waves</string>
```

**Review by ashutoshgngwr — COMMENTED** [review]
I can notice a pattern after listening to it for about two minutes. There's a regularity in the crashing wave, i.e. it crashes every ~6 seconds. I think it shouldn't be regular and crash after arbitrary intervals. I'd like to hear what you think about it.

Otherwise, this PR looks great, but unfortunately, I can't merge it (at least not right away). I'm currently working on v2, that effectively makes the sound library proprietary. I can add it to the v2 sound library, but it might take a while to release.

**Comment by Rult** [comment]
> There's a regularity in the crashing wave, i.e. it crashes every ~6 seconds. I think it shouldn't be regular and crash after arbitrary intervals. I'd like to hear what you think about it.

Maybe just include SoX in the app? That way raw noises will be smoother than just reusing mp3 samples and we could manipulate them better (see #842).

**Comment by ashutoshgngwr** [comment]
That will break the current flow of the sound engine, and the gain would be marginal. I am sure a little editing on a larger sample will get the job done. I'll look into it later and try to include a larger sample in the v2 sound library.

**Comment by Rult** [comment]
I think it could sound better if it reused the same volume pattern and length as [seaside.mp3](https://github.com/ashutoshgngwr/noice/blob/main/app/src/main/assets/seaside.mp3). Could you give a cue how I could do that? I'm not that good at audio editing yet.

**Comment by ashutoshgngwr** [comment]
@Rult

I generated a 5-minute sample using the command you provided. I reduced the wave frequency to 0.05 Hz. I got the following result:

image

As expected, each wave is slightly different from the other, but the gap between the two waves is almost equal. I was wondering if there's a way to vary the wait interval between waves, but I couldn't find anything relevant. I also noticed that the output is a single channel. I wonder if SoX can generate dual channel output.

You can use the following process for editing, but only if you _really want to_ edit it yourself. Otherwise, I will do it myself but sometime later.

1. Use [Audacity](https://www.audacityteam.org/download/) (because that's what I am familiar with).
2. Generate a long sample using SoX (5-10 mins).
3. Import the sample into Audacity.
4. Apply loudness normalization at **-27 LUFS** from the Effect menu before editing.
4. Remove the arbitrary length of a clip from the gaps.
   - Select a portion of the clip.
   - Delete it (AFAIK **Ctrl + K** or **Del**).
   - Play the clip and see if it has any discontinuity at the joint (usually audible as distinctive clicks).
   - If the click is audible at the joint, select 500ms portion around the joint (250ms on either side) and then apply the Crossfade Clips effect (you may need to enable it before it's visible in the effect menu).
   - Replay the sound and see if the click is gone.
   - Repeat until you have the desired output.
5. Export Audio as WAV (PCM 32-bit float) so that you can edit it later without worrying about quality loss.
6. Export Audio as MP3 128kbps or lower to include in the app.

You may find a better audio editing process on the Internet. I've laid out what works for me in general.

The final audio can be at most 2 minutes long.

**Comment by Rult** [comment]
Thanks! I'll leave this to you to ticker with until I get any ideas myself.

**Comment by ashutoshgngwr** [comment]
@Rult sorry for keeping this open for so long. I was busy with Noice's backend. I'm resuming work on the next version of Noice's Android client and will merge a version of this ASAP, albeit in a different (private sound library) repository.

**Comment by ashutoshgngwr** [comment]
I completely forgot about this, and perhaps for the better. There hasn't been any activity around this. So, I am going to close this.

Closes #842

**Pull request closed without merging**